### PR TITLE
[travis] Also run coqchk on HoTT

### PR DESCRIPTION
**Pull request by JasonGross**
I expect this to fail on travis; if it does, this is a bug in coqchk.  This should not be merged until coqchk is fixed.

### CODE REVIEW

File: dev/ci/ci-hott.sh
Comment (ejgallego) [inline dev/ci/ci-hott.sh]: Does this build HoTT twice?

Comment (JasonGross) [inline dev/ci/ci-hott.sh]: The `.vo` files aren't deleted in between, so nothing should be built twice.

Comment (ejgallego) [inline dev/ci/ci-hott.sh]: I meant the timing script, I am not sure we want that, and in that form.

I we go that route, I think the timing script should be standardized and merged into Coq, I am not comfortable relying on possibly out-of-sync scripts among Travis developments.

Comment (JasonGross) [inline dev/ci/ci-hott.sh]: Ah, well, if this target works before we come to a decision on #733, I can strip off the timing commit from this PR, and we can merge the validate commit without it.

Is the issue that you're worried that the build might fail because of a bad script and it'll be hard to diagnose?  What do you think of this alternative?
```bash
( cd ${HoTT_CI_DIR} && ./autogen.sh && ./configure && (make -j ${NJOBS} TIMED=1 | tee time-of-build.log; exit ${PIPESTATUS[0]}) && make validate )
python "./etc/coq-scripts/timing/make-one-time-file.py" time-of-build.log time-of-build-pretty.log && cat time-of-build-pretty.log
```
This way we separate the running of the test (which is done entirely by `make`) from the timing display (done by the submodule).

Comment (ejgallego) [inline dev/ci/ci-hott.sh]: I think I'd rather have the timing script merged in Coq and supported there; my worry is state-space explosion; even if it is unlikely, having an outside script running in the core Travis part introduces yet another factor to check when a problem arises.

Comment (JasonGross) [inline dev/ci/ci-hott.sh]: *nods*  I'd be happy to have it merged into Coq, though I'm not sure where it should go.  It's [a pretty simple python script](https://github.com/JasonGross/coq-scripts/blob/master/timing/TimeFileMaker.py); basically, it parses the output of one or two `make TIMED=1`, associates the time(s) to file names, takes the differences if there are two text blobs you pass it, sorts them, and prints them.

Where do you think it should go / how should it be called?

Comment (ejgallego) [inline dev/ci/ci-hott.sh]: `dev/tools` looks like a good place? Or maybe `coq_makefile` should be made aware of it if `COQ_TIMED=1` ?

That way we could uniformly build timings for all the CIs without even having to tweak the individual script files, just setting `COQ_TIMED` at the top-level travis file. I guess that this way the coq-bench infrastructure could also take profit from that more easily.

WDYT? OMWV! @gares?

Comment (ejgallego) [inline dev/ci/ci-hott.sh]: Indeed I think it makes sense to have `coq_makefile` output a kind of "official" timing format, and then allow the tools to compare one or more timing logs easily.

**Comment by ejgallego** [comment]
As I had guessed, this was not properly initializing the submodules, thus the build fails.

**Comment by JasonGross** [comment]
@ejgallego Are you sure the 8.7 milestone is right for this PR?  This blocks on [bug #5587](https://coq.inria.fr/bugs/show_bug.cgi?id=5587) (a bug in coqchk).

**Comment by ejgallego** [comment]
IMHO critical bugs as this one should of course try to be fixed ASAP, but yeah, I guess at some point it will become 8.8.

**Comment by JasonGross** [comment]
@ejgallego Is `fix_needed` the right label?  The "fix" that is needed is in Coq proper, in `coqchk`, not in this PR :-P  (Is there a separate label for `exposes_bug`.)

Also, I rebased this to discard the timing display

**Comment by ejgallego** [comment]
Well this cannot be merged as-is , it'd be great if we could signal dependencies, so maybe a `waiting_for`tag is warranted; for now it is either `fix_needed` or `in_progress`...

**Comment by SkySkimmer** [comment]
#1075 got merged.

**Comment by maximedenes** [comment]
It seems HoTT now fails with a `coqchk` error!

**Comment by JasonGross** [comment]
Indeed, that is what this PR was originally about, in some sense.  #1075 removed the issue that was hiding the output of the error message.

**Comment by maximedenes** [comment]
I see, there was a bit of confusion when the label was removed. We should certainly have a look soon at the bug. Thanks!

**Comment by Zimmi48** [comment]
Postponing once again. Has the bug that is bllocking this been reported independently?

**Comment by JasonGross** [comment]
@Zimmi48 No, it has not

**Comment by Zimmi48** [comment]
Maybe if it was, it would motivate someone to start working on it.

**Comment by Zimmi48** [comment]
@JasonGross Could you report the bug which is blocking this as a separate issue?

**Comment by JasonGross** [comment]
@Zimmi48 blocked by #6798

**Comment by SkySkimmer** [comment]
After #7317 and #7325 HoTT still doesn't go through:
>  checking cst:HoTT.Modalities.Modality.Modalities_Theory.RSU.isequiv_conn_map_ino
>  Fatal Error: Anomaly "Universe Var(0) undefined."
             Please report at http://coq.inria.fr/bugs/.

**Comment by SkySkimmer** [comment]
Currently depending on:
- #7317 #7325 #7328 (waiting for a checker maintainer to merge)
- #7329 (waiting for a fix)

**Comment by ppedrot** [comment]
@JasonGross You should probably rebase on a modern master.

**Comment by ejgallego** [comment]
Indeed, please rebase @JasonGross so we can see where we are at.

**Comment by ejgallego** [comment]
I rebased and pushed myself to https://gitlab.com/ejgallego/coq/pipelines/24895718 as I don't have write permissions to this branch.

**Comment by Zimmi48** [comment]
So indeed, this (https://gitlab.com/ejgallego/coq/-/jobs/78567664) still fails, with an anomaly:

```
Fatal Error: Anomaly "Uncaught exception Reduction.NotConvertible."
             Please report at http://coq.inria.fr/bugs/.
```

Is this one known or not?

**Comment by SkySkimmer** [comment]
#7329 isn't fixed yet.

**Comment by JasonGross** [comment]
>  as I don't have write permissions to this branch.

"Allow edits from maintainers" is checked...

In any case, I've force-pushed a rebase here, too.

**Comment by ejgallego** [comment]
@JasonGross can you rebase? One bug has been fixed in master.

**Comment by JasonGross** [comment]
Rebased

**Comment by ejgallego** [comment]
Hott seems ok now.

**Review by SkySkimmer — APPROVED** [review]

**Comment by Zimmi48** [comment]
VST seems broken everywhere. Clearly unrelated to this change anyways.

**Review by ejgallego — APPROVED** [review]
Failure is spurious, merging.

**Pull request merged**
